table/table_factory.cc

```diff
@@ -37,7 +37,8 @@ Status TableFactory::CreateFromString(const ConfigOptions& config_options_in,
     factory->reset(new CuckooTableFactory());
 #endif  // ROCKSDB_LITE
   } else {
-    return Status::NotSupported("Could not load table factory: ", name);
+    status = Status::NotSupported("Could not load table factory: ", name);
+    return status;
```

Comment:
Why this change is needed?

Reply:
If we reach here and proceed, the `status` obj created in line 17 will be unchecked.